pystac/asset.py

```diff
@@ -0,0 +1,165 @@
+from copy import copy
+from typing import Any, Dict, List, Optional, TYPE_CHECKING, Union
+
+import pystac as ps
+from pystac.utils import is_absolute_href, make_absolute_href
+
+if TYPE_CHECKING:
+    from pystac.collection import Collection as Collection_Type
+    from pystac.item import Item as Item_Type
+
+
+class Asset:
+    """An object that contains a link to data associated with an Item or Collection that
+    can be downloaded or streamed.
+
+    Args:
+        href (str): Link to the asset object. Relative and absolute links are both
```

Comment:
Can you drop the `(str)` type entry here an below if there are type annotations?  e.g. what will ipython's `foo?` give?  I think it gives the type annotations, so these are redundant.

Reply:
Tracking in #323